src/main/java/com/target/studentinfo/controller/StudentController.java

```diff
@@ -20,16 +20,29 @@ public StudentController(StudentService studentService) {
     }
 
     @GetMapping("/students")
-    public List<Student> getAllStudentInfo(){
+    public List<Student> getAllStudentInfo(@RequestParam(defaultValue="false") Boolean isAdmin){
 
-        return studentService.getAllStudentInfo();
+        if(isAdmin==true)
```

Comment:
Do not use the name `isAdmin` just yet. We may not implement the role-access functionality. Maybe `includeSoftDeletes` ?  We'll discuss and rename.